### PR TITLE
Stefan jum/add dev content

### PR DESCRIPTION
**Pull request by StefanJum**

### CODE REVIEW

**Comment by razvand** [comment]
Please also add an update to the `README.md` file. See the updates from [this commit](https://github.com/unikraft/app-elfloader/commit/450a52101b17ed041ae97b9d219c997669309cdf).

Also update `.unikraft/` to `workdir/` and use the proper names for defconfig files.

**Comment by StefanJum** [comment]
Done @razvand

File: README.md [hunk 1]
```diff
@@ -229,6 +229,47 @@ Follow the steps below for the setup:
      10 directories, 7 files
      ```
 
+## Scripted Building and Running
+
+To make it easier build and run and test different configurations, the repository provides a set of scripts that do everything required.
```
Comment (razvand) [inline README.md]:
```suggestion
To make it easier to build, run and test different configurations, the repository provides a set of scripts that do everything required.
```

File: README.md [hunk 2]
```diff
@@ -229,6 +229,47 @@ Follow the steps below for the setup:
...
+These are scripts used for building different configurations of the SQLite server and for running these with all the requirements behind the scenes.
+
+First, generate these scripts by running the `generate.py` script while on the `sqlite` directory:
```
Comment (razvand) [inline README.md]:
```suggestion
First of all, grab the [`generate.py` script](https://github.com/unikraft/app-testing/blob/staging/scripts/generate.py) and place it in the `scripts/` directory by running:

'''console
wget https://raw.githubusercontent.com/unikraft/app-testing/staging/scripts/generate.py -O scripts/generate.py
chmod a+x scripts/generate.py
'''

Now, run the `generate.py` script.
You must run it in the root directory of this repository:
```

Replace the apostrophes above with backticks.

**Review by razvand — CHANGES_REQUESTED** [review]
Remove the `generate.py` script, since it's now downloaded. Also ignore it.

Also update the TLDR in `README.md` with the downloading and using of `generate.py`, same as documented [here](https://github.com/unikraft/app-llama2-c/#quick-set-up-aka-tldr).

**Review by razvand — COMMENTED** [review]
Also rename `fs0/` to `rootfs/`.

Ignore `generate.py` in `scripts/`. I recommend a `.gitignore` file in `scripts/`.

File: defconfigs/fc-x86_64-initrd [hunk 3]
```diff
@@ -0,0 +1,10 @@
+CONFIG_PLAT_KVM=y
```
Comment (razvand) [inline defconfigs/fc-x86_64-initrd:1]: I would add CONFIG_UK_DEFNAME și CONFIG_UK_NAME to each defconfig file.

File: scripts/setup.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (razvand) [inline scripts/setup.py]: Why is it called `setup.py`? Shouldn't it bee `setup.sh`?

File: scripts/run.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+runs:
+  - name: sqlite
```
Comment (razvand) [inline scripts/run.yaml]:
```suggestion
```

File: scripts/run.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+    rootfs: ./fs0
+    command: ""
```
Comment (razvand) [inline scripts/run.yaml]:
```suggestion
```

File: scripts/run.yaml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+runs:
+  - name: sqlite
+    rootfs: ./fs0
```
Comment (razvand) [inline scripts/run.yaml]: Rename this to `rootfs/`.

**Review by razvand — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 8]
```diff
@@ -3,3 +3,4 @@
 /build/
 /.unikraft/
 /fs0.cpio
+/workdir/
```
Comment (razvand) [inline .gitignore:5]:
```suggestion
/workdir/
/.config*
```

File: .gitignore [hunk 9]
```diff
@@ -3,3 +3,4 @@
 /build/
 /.unikraft/
 /fs0.cpio
```
Comment (razvand) [inline .gitignore]:
```suggestion
/rootfs.cpio
```

**Review by razvand — CHANGES_REQUESTED** [review]
You took out `fs0/` by mistake. Rename it to `rootfs/`.

Rename `aarch64` to `arm64` in `kraft.yaml` target names.

File: scripts/setup.sh [hunk 10]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+[ ! -d "workdir" ] && echo "Cloning unikraft..." || true
```
Comment (razvand) [inline scripts/setup.sh]:
```suggestion
test ! -d "workdir" && echo "Cloning repositories ..." || true
```

File: scripts/setup.sh [hunk 11]
```diff
@@ -0,0 +1,6 @@
...
+[ ! -d "workdir/unikraft" ] && git clone https://github.com/unikraft/unikraft workdir/unikraft || true
```
Comment (razvand) [inline scripts/setup.sh]:
```suggestion
test ! -d "workdir/unikraft" && git clone https://github.com/unikraft/unikraft workdir/unikraft || true
```

File: scripts/setup.sh [hunk 12]
```diff
@@ -0,0 +1,6 @@
...
+[ ! -d "workdir/libs/sqlite" ] && git clone https://github.com/unikraft/lib-sqlite workdir/libs/sqlite || true
```
Comment (razvand) [inline scripts/setup.sh]:
```suggestion
test ! -d "workdir/libs/sqlite" && git clone https://github.com/unikraft/lib-sqlite workdir/libs/sqlite || true
```

File: scripts/setup.sh [hunk 13]
```diff
@@ -0,0 +1,6 @@
...
+[ ! -d "workdir/libs/musl" ] && git clone https://github.com/unikraft/lib-musl workdir/libs/musl || true
```
Comment (razvand) [inline scripts/setup.sh]:
```suggestion
test ! -d "workdir/libs/musl" && git clone https://github.com/unikraft/lib-musl workdir/libs/musl || true
```

**Review by razvand — CHANGES_REQUESTED** [review]
Use `test` instead of `[`. It's more readable and it's consistent with other scripts we're using.

**Review by razvand — APPROVED** [review]
Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**